src/portable/server/list_versions.rs

```diff
@@ -1,96 +1,187 @@
-                    package: None,
-                    install: None,
-                })
-                .install
-                .insert(install);
-        }
-        if cmd.json {
-            print!(
-                "{}",
-                serde_json::to_string_pretty(
-                    &version_set
-                        .into_iter()
-                        .map(|(ver, vp)| JsonVersionInfo {
-                            channel: Channel::from_version(&ver).unwrap_or(Channel::Nightly),
-                            version: vp.install.as_ref().map_or_else(
-                                || vp.package.as_ref().unwrap().version.clone(),
-                                |v| v.version.clone(),
-                            ),
-                            installed: vp.install.is_some(),
-                            debug_info: DebugInfo {
-                                install: vp.install.map(DebugInstall::from),
-                                package: vp.package,
-                            },
-                        })
-                        .collect::<Vec<_>>()
-                )?
-            );
+        // Combine the channel, version and installed_only into a single string of plain english:
+        log::debug!(
+            "channel: {:?}, version: {:?}, installed_only: {:?}",
+            channel,
+            version,
+            cmd.installed_only
+        );
+
+        let filter = if cmd.installed_only {
+            let mut filter = "installed".to_string();
+            if version.is_some() {
+                filter.push_str(&format!(", matching"));
+            }
+            if let Some(channel) = channel {
+                filter.push_str(&format!(", {channel:?} channel").to_lowercase());
+            }
+            filter
         } else {
-            print_table(version_set.into_values().map(|vp| match vp.install {
-                Some(v) => (v.version, true),
-                None => (vp.package.unwrap().version, false),
-            }));
-        }
+            let mut filter = if cmd.version.is_some() {
+                "matching, "
+            } else if channel == Some(Channel::Stable) {
+                "current, "
+            } else {
+                ""
+            }
+            .to_string();
+            if let Some(channel) = channel {
+                filter.push_str(&format!("{channel:?} channel").to_lowercase());
+            }
+            filter
+        };
+
+        let time = if cmd.version.is_none() {
+            "current"
+        } else {
+            "matching"
+        };
+        let channel = if cmd.channel.is_some() {
+            format!("{:?}", cmd.channel.unwrap()).to_lowercase()
+        } else {
+            "".to_string()
+        };
+
+        let installed = if cmd.installed_only {
+            "installed"
+        } else {
+            "all"
+        };
+
+        cprintln!("Listing <bold>{filter}</bold> versions of <bold>{BRANDING}</bold>:");
```

Comment:
When listing only current stable versions, can we also inject a note on how to expand the list, e.g `use --all to list all available versions or --channel=<channel> to list versions available in another release channel`).